### ISSUE TITLE
external service override host header

### ISSUE THREAD

**Issue by skaravad** [kumahq/kuma#3921]
### Description

When we create external services, services inside mesh can use the service name defined in the tags 
```apiVersion: kuma.io/v1alpha1
kind: ExternalService
mesh: default
metadata:
  name: httpbin
spec:
  tags:
    kuma.io/service: httpbin
    kuma.io/protocol: http
  networking:
    address: httpbin.org:443
    tls: # optional
      enabled: true
      allowRenegotiation: false
      sni: httpbin.org # optional
```
but this will not work if the actual external service is strict about the host header , for instance use google.com in `networking.address` and then try `curl httpbin` , this will result in google not understanding the call at all, this is the same case with 90% of the services where traffic to destinations is controlled by host header.

One way to workaround the issue is to use actual hostname defined in the `networking.address` when making the calls. If the host header can be set by the external service itself, then I think it should just work, the `sni` field in the `tls` section is one such key, though today it is used for a different reason.

To summarise: `mesh service -> external service -> set host header -> destination` this something reverse proxies provide to override the incoming host header when communicating with backend, something like nginx does
```
proxy_set_header Host actual_host;
```

**Comment by lahabana** [comment]
We should not reuse SNI for this.

```yaml
spec:
   networking:
      http:
        host: google.com
```

For other use cases may need traffic route or needs more complex design.

**Comment by lahabana** [comment]
For triage: I feel like this has been done already no?

**Comment by kleinfreund** [comment]
**Triage**: We think this is possible using proxy templates and we’re going to check that.

**Comment by lobkovilya** [comment]
I think it's implemented here https://github.com/kumahq/kuma/pull/1635

**Comment by lahabana** [comment]
@bartsmykla is this a dupe of what you are working #5895

**Comment by bartsmykla** [comment]
It's not. I'm working on rewrites for builtin gateway routes. But as @lobkovilya mentioned, auto_host_rewrite is the default value currently for external services

**Comment by bartsmykla** [comment]
I'm closing this as I believe this issue was already solved.
To test it I applied policy below:
```sh
echo 'apiVersion: kuma.io/v1alpha1
kind: ExternalService
mesh: default
metadata:
  name: google
spec:
  tags:
    kuma.io/service: google
    kuma.io/protocol: http
  networking:
    address: google.com:443
    tls:
      enabled: true
' | kubectl apply -f-
```
then checked requests from one of the pods belonging to the mesh
```sh
/ # curl google.com
<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>301 Moved</TITLE></HEAD><BODY>
<H1>301 Moved</H1>
The document has moved
<A HREF="https://www.google.com/">here</A>.
</BODY></HTML>

/ # curl google.mesh
<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>301 Moved</TITLE></HEAD><BODY>
<H1>301 Moved</H1>
The document has moved
<A HREF="https://www.google.com/">here</A>.
</BODY></HTML>
```

@skaravad please reopen it if you are still encountering the issue
